### DISCUSSION TITLE
RTL-SDR setup assistance

### DISCUSSION THREAD

**Discussion by Sandman2001** [merbanan/rtl_433#3025] [Q&A]
I recently purchased a [RTL blog V4](https://www.rtl-sdr.com/rtl-sdr-blog-v4-dongle-initial-release/) and working on configuring to connect to a WH65B (previous discussion #3002). I am running on [WSL's ](https://learn.microsoft.com/en-us/windows/wsl/) newest version. I have installed SoapySDR and binded the usb to the linux terminal through powershell. `sandera2@DESKTOP-P1P04CT:~$ SoapySDRUtil --find`
######################################################
######################################################

[INFO] [UHD] linux; GNU C++ version 11.2.0; Boost_107400; UHD_4.1.0.5-3
Found Rafael Micro R828D tuner
Found device 0
  default_input = True
  default_output = False
  device_id = 0
  driver = audio
  label = RDP Source

Found device 1
  default_input = False
  default_output = False
  device_id = 2
  driver = audio
  label = Monitor of RDP Sink

Found device 2
  driver = rtlsdr
  label = Generic RTL2832U OEM :: 00000001
  manufacturer = RTLSDRBlog
  product = Blog V4
  serial = 00000001
  tuner = Rafael Micro R828D
`

**Installation steps for RTL**
first started with `sudo apt-get install rtl-433`
went into build folder and tried following the steps in [this video](https://www.youtube.com/watch?time_continue=260&v=A_aqA377i2w&embeds_referring_euri=https%3A%2F%2Fwww.bing.com%2F&embeds_referring_origin=https%3A%2F%2Fwww.bing.com&source_ve_path=MTM5MTE3LDEzOTExNywxMzkxMTcsMjg2NjY)

got this:

`sandera2@DESKTOP-P1P04CT:~/rtl_433/build$ sudo make uninstall
[sudo] password for sandera2:
CMake Error at cmake_uninstall.cmake:4 (MESSAGE):
  Cannot find install manifest:
  "/home/sandera2/rtl_433/build/install_manifest.txt"

make[3]: *** [CMakeFiles/uninstall.dir/build.make:70: CMakeFiles/uninstall] Error 1
make[2]: *** [CMakeFiles/Makefile2:197: CMakeFiles/uninstall.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:204: CMakeFiles/uninstall.dir/rule] Error 2
make: *** [Makefile:179: uninstall] Error 2
sandera2@DESKTOP-P1P04CT:~/rtl_433/build$ sudo make clean
[sudo] password for sandera2:
sandera2@DESKTOP-P1P04CT:~/rtl_433/build$ sudo cmake
Usage

  cmake [options] <path-to-source>
  cmake [options] <path-to-existing-build>
  cmake [options] -S <path-to-source> -B <path-to-build>

Specify a source directory to (re-)generate a build system for it in the
current working directory.  Specify an existing build directory to
re-generate its build system.

Run 'cmake --help' for more information.

sandera2@DESKTOP-P1P04CT:~/rtl_433/build$ sudo cmake ..
CMake Deprecation Warning at CMakeLists.txt:4 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.

-- Build type not specified: defaulting to release.
-- Found Git version: refs/heads/master commit 09d62510a42951a1fb46f1782a22185a90e331b1 from 2024-08-06T18:40:57+0200
-- Using Git version tag: 23.11-155-g09d62510 on master at 202408061840
-- IPv6 support enabled.
-- Threads support will be compiled.
-- Could NOT find OpenSSL, try to set the path to OpenSSL root folder in the system variable OPENSSL_ROOT_DIR (missing: OPENSSL_CRYPTO_LIBRARY OPENSSL_INCLUDE_DIR)
-- OpenSSL development files not found, TLS won't be possible.
-- RTL-SDR device input will be compiled. Found version 0.6.0
-- libusb-1.0 error messages are available. Found version 1.0.25
-- SoapySDR device input will be compiled. Found version 0.8.1-PothosSDR-2021.07.25-vc16-x64
-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE)
-- Configuring done
-- Generating done
-- Build files have been written to: /home/sandera2/rtl_433/build
sandera2@DESKTOP-P1P04CT:~/rtl_433/build$ sudo make
[  1%] Building C object src/CMakeFiles/r_433.dir/abuf.c.o
[  1%] Building C object src/CMakeFiles/r_433.dir/am_analyze.c.o
[  1%] Building C object src/CMakeFiles/r_433.dir/baseband.c.o
[  2%] Building C object src/CMakeFiles/r_433.dir/bit_util.c.o
[  2%] Building C object src/CMakeFiles/r_433.dir/bitbuffer.c.o
[  2%] Building C object src/CMakeFiles/r_433.dir/compat_paths.c.o
[  3%] Building C object src/CMakeFiles/r_433.dir/compat_time.c.o
[  3%] Building C object src/CMakeFiles/r_433.dir/confparse.c.o
[  3%] Building C object src/CMakeFiles/r_433.dir/data.c.o
/home/sandera2/rtl_433/src/data.c:156:32: warning: unknown option after ‘#pragma GCC diagnostic’ kind [-Wpragmas]
  156 | #pragma GCC diagnostic ignored "-Wunknown-warning-option"
      |                                ^~~~~~~~~~~~~~~~~~~~~~~~~~
[  4%] Building C object src/CMakeFiles/r_433.dir/data_tag.c.o
[  4%] Building C object src/CMakeFiles/r_433.dir/decoder_util.c.o
[  4%] Building C object src/CMakeFiles/r_433.dir/fileformat.c.o
[  5%] Building C object src/CMakeFiles/r_433.dir/http_server.c.o
[  5%] Building C object src/CMakeFiles/r_433.dir/jsmn.c.o
[  5%] Building C object src/CMakeFiles/r_433.dir/list.c.o
[  6%] Building C object src/CMakeFiles/r_433.dir/logger.c.o
[  6%] Building C object src/CMakeFiles/r_433.dir/mongoose.c.o
[  6%] Building C object src/CMakeFiles/r_433.dir/optparse.c.o
[  7%] Building C object src/CMakeFiles/r_433.dir/output_file.c.o
[  7%] Building C object src/CMakeFiles/r_433.dir/output_influx.c.o
[  7%] Building C object src/CMakeFiles/r_433.dir/output_log.c.o
[  8%] Building C object src/CMakeFiles/r_433.dir/output_mqtt.c.o
[  8%] Building C object src/CMakeFiles/r_433.dir/output_rtltcp.c.o
In file included from /home/sandera2/rtl_433/include/compat_pthread.h:39,
                 from /home/sandera2/rtl_433/src/output_rtltcp.c:20:
/mnt/c/Program Files/PothosSDR/include/pthread.h:108: warning: "PTW32_LEVEL" redefined
  108 | #define PTW32_LEVEL PTW32_LEVEL_MAX
      |
/mnt/c/Program Files/PothosSDR/include/pthread.h:95: note: this is the location of the previous definition
   95 | #define PTW32_LEVEL 1
      |
In file included from /mnt/c/Program Files/PothosSDR/include/pthread.h:303,
                 from /home/sandera2/rtl_433/include/compat_pthread.h:39,
                 from /home/sandera2/rtl_433/src/output_rtltcp.c:20:
/mnt/c/Program Files/PothosSDR/include/sched.h:64: warning: "PTW32_LEVEL" redefined
   64 | #define PTW32_LEVEL PTW32_LEVEL_MAX
      |
/mnt/c/Program Files/PothosSDR/include/sched.h:51: note: this is the location of the previous definition
   51 | #define PTW32_LEVEL 1`

much more errors not listed from same compile
end of error codes: 
`/home/sandera2/rtl_433/src/output_rtltcp.c:562: warning: control reaches end of non-void function [-Wreturn-type]
make[2]: *** [src/CMakeFiles/r_433.dir/build.make:384: src/CMakeFiles/r_433.dir/output_rtltcp.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:951: src/CMakeFiles/r_433.dir/all] Error 2
make: *** [Makefile:146: all] Error 2`

**Running RTL**

i can access the help command from the cli and run stuff, but no sensor data returns

commands I am trying to enter:

`sandera2@DESKTOP-P1P04CT:~/rtl_433$ rtl_433 -v -F json -p 39.741 -R 78 -f 914980000 -s 250000 -vv
rtl_433 version 21.12 (2021-12-14) inputs file rtl_tcp RTL-SDR SoapySDR
Use -h for usage help and see https://triq.org/ for documentation.
Trying conf file at "rtl_433.conf"...
Trying conf file at "/home/sandera2/.config/rtl_433/rtl_433.conf"...
Trying conf file at "/usr/local/etc/rtl_433/rtl_433.conf"...
Trying conf file at "/etc/rtl_433/rtl_433.conf"...
Registering protocol [78] "Fine Offset Electronics, WH25, WH32B, WH24, WH65B, HP1000 Temperature/Humidity/Pressure Sensor"

New defaults active, use "-Y classic -s 250k" for the old defaults!

Registered 1 out of 207 device decoding protocols
Found 1 device(s)

trying device  0:  RTLSDRBlog, Blog V4, SN: 00000001
Found Rafael Micro R828D tuner
Using device 0: Generic RTL2832U OEM
Exact sample rate is: 250000.000414 Hz
[R82XX] PLL not locked!
Sample rate set to 250000 S/s.
Bit detection level set to 0.0 (Auto).
Tuner gain set to Auto.
Frequency correction set to 39 ppm.
Reading samples in async mode...
Tuned to 914.980MHz.
Allocating 15 zero-copy buffers
baseband_demod_FM: low pass filter for 250000 Hz at cutoff 50000 Hz, 20.0 us
^CSignal caught, exiting!`

currently stuck on what to address

**Comment by zuckschwerdt** [comment]
Did you try a binary [release](https://github.com/merbanan/rtl_433/releases)? You don't need SoapySDR for an RTL-SDR and you likely don't need WSL.

Seems like threads support didn't work when building in WSL (the output_rtltcp.c warning is a clue. not sure though why that return NULL is skipped). But you didn't use that build anyway then :)

**Reply by Sandman2001** [reply]
![image](https://github.com/user-attachments/assets/4a25b365-a3bc-48d9-ba2f-572444cf80a0)
still stuck
p.s. I adjusted sample rate to 250k from 25k and still no change

**Reply by zuckschwerdt** [reply]
Try to get more reporting with just `rtl_433 -Y autolevel -Y magest -M level -M noise -M protocol`

But really I guess that the librtlsdr.dll is not updated for the Blog v4. Find a DLL updated for the RTL blog V4 and replace it in that dir. Should then work as expected.

**Reply by Sandman2001** [reply]
![image](https://github.com/user-attachments/assets/508bb843-e600-42f7-b215-483b60d2ef47)
I wasn't able to reproduce the LIBUSB error shown above, but I adjusted the antenna while running the program to see if anything would change and confirmed some response as seen in the last debugging run. Still have yet to update the .dll files so I will let you know if anything changes with that.

**Reply by zuckschwerdt** [reply]
The levels look like an active band. Noise levels above -15dB are not good, might be a busy band or a bad antenna, levels below -30 dB would be dead silence.
